From 0c1257b50c4d0af20d20d289ceb6e79ba2cf490a Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Maxime=20M=C3=A9r=C3=A9?= <maxime.mere@foss.st.com>
Date: Mon, 25 Nov 2024 11:23:44 +0100
Subject: [PATCH 1366/1695] drivers: stm32_rng: enable all oscillator in
 STM32MP21
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Set the NSCR value to enable all oscillators. This improves RNG entropy.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git 3acdc8cc7010464d021cd6194143d2b71114aeff drivers: stm32_rng: enable all oscillator in STM32MP21]
Signed-off-by: Maxime Méré <maxime.mere@foss.st.com>
Change-Id: I757e587c5fc10314c64a0c76bf0bdf4b4382f551
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/420539
Domain-Review: Yann GAUTIER <yann.gautier@foss.st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
Reviewed-by: Thomas BOURGOIN <thomas.bourgoin@foss.st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
---
 core/drivers/stm32_rng.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/core/drivers/stm32_rng.c b/core/drivers/stm32_rng.c
index ab5bebfc5..80ae7edba 100644
--- a/core/drivers/stm32_rng.c
+++ b/core/drivers/stm32_rng.c
@@ -734,7 +734,7 @@ static const struct stm32_rng_driver_data mp21_data[] = {
 		.has_cond_reset = true,
 		.has_power_optim = true,
 		.cr = 0x00800D00,
-		.nscr = 0x01F7,
+		.nscr = 0x01FF,
 		.htcr = 0xAAC7,
 	},
 };
-- 
2.47.2

